pkg/lib/indexer/indexer.go

```diff
@@ -253,15 +290,27 @@ func (i ImageIndexer) PruneFromIndex(request PruneFromIndexRequest) error {
 		return err
 	}
 
-	databasePath, err := i.ExtractDatabase(buildDir, request.FromIndex, request.CaFile, request.SkipTLS)
+	fromIndex := image.SimpleReference(request.FromIndex)
+
+	// Get the old index image's dbLocationLabel to find this path
+	labels, err := i.ImgRegistry.Labels(context.TODO(), fromIndex)
+	if err != nil {
+		return err
+	}
+
+	dbLocation, ok := labels[containertools.DbLocationLabel]
+	if !ok {
+		return fmt.Errorf("index image %s missing label %s", request.FromIndex, containertools.DbLocationLabel)
+	}
+	extractedDbLocation, err := i.ExtractIndexContent(fromIndex, dbLocation, filepath.Join(buildDir, defaultDatabaseFolder))
```

Comment:
It looks like this section of code is repeated in several places. Can we put it in a separate function?